src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/ServiceIndexResourceV3.cs

```diff
@@ -14,129 +16,210 @@ namespace NuGet.Protocol
     /// </summary>
     public class ServiceIndexResourceV3 : INuGetResource
     {
-        private readonly IDictionary<string, List<Uri>> _index;
+        private readonly IDictionary<string, List<ServiceIndexEntry>> _index;
         private readonly DateTime _requestTime;
-        private List<Uri> _empty;
+        private static readonly IReadOnlyList<ServiceIndexEntry> _emptyEntries = new List<ServiceIndexEntry>();
```

Comment:
suggestion: Some comments on each of these would be nice to have. Else it is harder to understand.